### ISSUE TITLE
`read_csv_glob` fails with empty parse_date list

### ISSUE THREAD

**Issue by c3-cjazra** [modin-project/modin#4056]
### System information
Ray 1.9.2
Modin 0.12.0
Python 3.9.7

### Describe the problem

```
pd_experimental.read_csv_glob("somevalidpath", parse_dates=[])
----
...
File "##/python3.9/site-packages/modin/experimental/pandas/io.py", line 208, in _read
pd_obj = FactoryDispatcher.read_csv_glob(**kwargs)
File "##/python3.9/site-packages/modin/core/execution/dispatching/factories/dispatcher.py", line 185, in read_csv_glob
return cls.__factory._read_csv_glob(**kwargs)
File "##/python3.9/site-packages/modin/core/execution/dispatching/factories/factories.py", line 513, in _read_csv_glob
return cls.io_cls.read_csv_glob(**kwargs)
File "##/python3.9/site-packages/modin/core/io/text/csv_glob_dispatcher.py", line 238, in _read
if isinstance(parse_dates, list) and isinstance(parse_dates[0], list):
IndexError: list index out of range (worker#1713)
File "##/python3.9/site-packages/ray/worker.py", line 1713, in get
raise value.as_instanceof_cause()
```

it looks like code only checks that the type of `parse_dates` is a list, but doesn’t check for empty content, tries to access first element and throws an error

**Comment by mvashishtha** [comment]
@c3-cjazra Thank you! I have verified the bug on my computer. I'm working on a fix.

**Comment by c3-cjazra** [comment]
is this fixed in `0.14.0`?

**Comment by YarShev** [comment]
@c3-cjazra, yes, this is fixed in 0.14.0.

**Comment by devin-petersohn** [comment]
HI @c3-cjazra , for the future we published very detailed release notes. Here are the release notes for 0.14.0: https://github.com/modin-project/modin/releases/tag/0.14.0

Please let us know if there's any more detail you would want to see!
